File: Java_Programmers/1단계/src/음양더하기.java
public class 음양더하기 {
    public int solution(int[] absolutes, boolean[] signs) {
        int answer = 0;
        for (int i = 0; i < absolutes.length; i++) {
            answer += absolutes[i] * (signs[i] ? 1 : -1);
        }
        return answer;
    }

    public static void main(String[] args) {
        음양더하기 sol = new 음양더하기();
        int[] absolutes = {4, 7, 12};
        boolean[] signs = {true, false, true};
        System.out.println(sol.solution(absolutes, signs));
    }
}

File: python/python_Programmers/1단계/음양더하기.py
# 쉬운 풀이
def solution(absolutes, signs):
    answer = 0
    for i in range(0, len(absolutes)):
        if signs[i]:
            answer += abs(absolutes[i])
        else:
            answer -= abs(absolutes[i])
    return answer

# 삼항연산자 풀이
def solution2(absolutes, signs):
    return sum(absolutes if sign else -absolutes for absolutes, sign in zip(absolutes, signs))


print(solution([1,2,3], [False, False, True]))
print(solution2([1,2,3], [False, False, True]))